libs/asset-viewer/src/lib/state/asset-search/asset-search.selector.ts

```diff
@@ -0,0 +1,382 @@
+import { createSelector } from '@ngrx/store';
+import * as A from 'fp-ts/Array';
+import { pipe } from 'fp-ts/function';
+import * as O from 'fp-ts/Option';
+
+import { fromAppShared } from '@asset-sg/client-shared';
+import {
+  AssetEditDetail,
+  Contact,
+  DateRange,
+  GeometryCode,
+  LV95,
+  LineString,
+  Point,
+  ReferenceData,
+  Study,
+  StudyPolygon,
+  UsageCode,
+  ValueCount,
+  ValueItem,
+  ordStatusWorkByDate,
+  usageCodes,
+} from '@asset-sg/shared';
+
+
+import { AssetDetail } from '../../models';
+
+import { AppStateWithAssetSearch } from './asset-search.reducer';
+
+
+const assetSearchFeature = (state: AppStateWithAssetSearch) => state.assetSearch;
+
+export const selectMapInitialised = createSelector(assetSearchFeature, state => state.mapInitialised);
+
+export const selectIsRefineAndResultsOpen = createSelector(assetSearchFeature, state => state.isRefineAndResultsOpen);
+
+
+export const selectAssetSearchState = createSelector(assetSearchFeature, state => state);
+
+export const selectSearchLoadingState = createSelector(assetSearchFeature, state => state.searchLoadingState);
+
+export const selectAssetDetailLoadingState = createSelector(assetSearchFeature, state => state.assetDetailLoadingState);
+
+export const selectAssetSearchQuery = createSelector(assetSearchFeature, state => state.searchQuery);
+
+export const selectAssetSearchResultData = createSelector(assetSearchFeature, state => state.searchResults.data);
+
+export const selectAssetSearchPageData = createSelector(assetSearchFeature, state => state.searchResults.page);
+
+export const selectAssetSearchPolygon = createSelector(assetSearchFeature, state => state.searchQuery.polygon);
+
+export const selectStudies = createSelector(
+  selectAssetSearchResultData,
+  (assetEditDetails): StudyVM[] =>
+    assetEditDetails.flatMap(assetEditDetail => assetEditDetail.studies.map(study => {
+      return {
+        assetId: study.assetId,
+        studyId: study.studyId,
+        geom: wktToGeoJSON(study.geomText),
+      };
+    })),
+);
+
+export const selectAssetsSearchStats = createSelector(assetSearchFeature, state => state.searchStats);
+
+export const selectCurrentAssetDetail = createSelector(assetSearchFeature, state => state.currentAsset,
+);
+
+export const selectCurrentAssetDetailVM = createSelector(fromAppShared.selectRDReferenceData, fromAppShared.selectLocale, selectCurrentAssetDetail, (referenceData, locale, currentAssetDetail) => {
+  if (RD.isSuccess(referenceData) && !!currentAssetDetail) {
+    return makeAssetDetailVMNew(referenceData.value, currentAssetDetail, locale);
+  }
+  return null as ReturnType<typeof makeAssetDetailVMNew> | null;
+});
+
+export const selectDrawerState = createSelector(
+  selectMapInitialised,
+  selectIsRefineAndResultsOpen,
+  fromAppShared.selectIsPanelOpen,
+  selectCurrentAssetDetail,
+  (mapInitialised, isRefineAndResultsOpen, isPanelOpen, currentAssetDetail): DrawerState =>
```

Comment:
Prefix boolean variables with `is` or `has`.
```suggestion
  (isMapInitialised, isRefineAndResultsOpen, isPanelOpen, currentAssetDetail): DrawerState =>
```